pandas/_libs/properties.pyx

```diff
@@ -23,12 +23,29 @@ cdef class CachedProperty:
             return self
 
         # Get the cache or set a default one if needed
-        cache = getattr(obj, '_cache', None)
+        cache = instance_cache = getattr(obj, '_cache', None)
```

Comment:
why is any of this necessary?

Reply:
I didn't really know how to test this and implemented this runtime check. I think it's not absolutely necessary if one follows the convention properly but I believe it is very easy to make a mistake here.

Reply:
```

def test_no_cache_attribute_on_class():
   for cls in list_of_all_relevant_classes_somehow():
       assert not hasattr(cls, "_cache")
```